From: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
Date: Thu, 1 Dec 2011 19:09:43 +0900
Patch-mainline: no, we abort instead
References: FATE#306586
Subject: [PATCH] Btrfs: forced readonly when free_log_tree fails

The filesystem turns to readonly instead of BUG_ON() when
free_log_tree fails.

Signed-off-by: Tsutomu Itoh <t-itoh@jp.fujitsu.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/tree-log.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/fs/btrfs/tree-log.c
+++ b/fs/btrfs/tree-log.c
@@ -2471,7 +2471,10 @@ static void free_log_tree(struct btrfs_t
 
 	if (trans) {
 		ret = walk_log_tree(trans, log, &wc);
-		BUG_ON(ret);
+		if (ret) {
+			btrfs_std_error(log->fs_info, ret);
+			return;
+		}
 	}
 
 	while (1) {
